Uplift/Models/Gym.swift

```diff
@@ -127,4 +139,72 @@ struct DailyGymHours {
             dayOfWeek = 0
         }
     }
+
+    init(gymHoursDataId: AllGymsQuery.Data.Gym.Facility.Time?) {
+        if let gymHoursData = gymHoursDataId {
+            dayOfWeek = gymHoursData.day
+            openTime = Date.getTimeFromString(datetime: gymHoursData.startTime)
+            closeTime = Date.getTimeFromString(datetime: gymHoursData.endTime)
+        } else {
+            openTime = Date()
+            closeTime = openTime
+            dayOfWeek = 0
+        }
+    }
+}
+
+struct Facility {
+    var equipment: [Equipment]
+    var miscInformation: [String]
+    var name: String
+    var times: [DailyGymHours]
+
+    init(gymData: AllGymsQuery.Data.Gym.Facility) {
+        if let equipmentGymData = gymData.equipment {
+            equipment = equipmentGymData.compactMap({ (equipmentData) -> Equipment? in
+                guard let equipmentData = equipmentData else { return nil }
+                return Equipment(equipmentData: equipmentData)
+            })
+        } else {
+            equipment = []
+        }
+
+        var miscInformationList: [String] = []
+        if let informationData = gymData.miscInformation {
+            informationData.forEach({ (info) in
+                if let info = info {
+                    miscInformationList.append(info)
+                }
+            })
+        }
+        miscInformation = miscInformationList
+
+        name = gymData.name
+
+        times = gymData.times.compactMap({ (gymHoursDataId) -> DailyGymHours? in
+            guard let gymHoursDataId = gymHoursDataId else { return nil }
+            return DailyGymHours(gymHoursDataId: gymHoursDataId)
+        })
+    }
+}
+
+struct Equipment {
+    var equipmentType: String
+    var name: String
+    var quantity: String
+    var workoutType: String
+
+    init(equipmentData: AllGymsQuery.Data.Gym.Facility.Equipment?) {
+        if let equipmentData = equipmentData {
```

Comment:
kind of confused on this; can't you just use the nil coalescing to set everything to an empty string if equipmentData is nil (and for that matter, why does the else set quantity to "" but then quantity inside the if let is nil coalesced to "0" (which I don't think would ever be run since you defined quantity as a non-optional string in struct Equipment).

Reply:
(these most likely have been updated on master and going to be re-done with the new backend model)